spec/datadog/opentelemetry/metrics_spec.rb

```diff
@@ -0,0 +1,368 @@
+        c.version = '1.0.0'
+        c.env = 'test'
+        c.tags = {'team' => 'backend', 'region' => 'us-east-1', 'host.name' => 'myhost'}
+        c.tracing.report_hostname = true
+      end
+
+      expect(attributes['service.name']).to eq('test-service')
+      expect(attributes['service.version']).to eq('1.0.0')
+      expect(attributes['deployment.environment']).to eq('test')
+      expect(attributes['host.name']).to eq("myhost")
+      expect(attributes['team']).to eq('backend')
+      expect(attributes['region']).to eq('us-east-1')
+    end
+
+    it 'applies fallback service name when neither DD_SERVICE nor service tag is set' do
+      setup_metrics
+
+      expect(attributes['service.name']).to eq(Datadog::Core::Environment::Ext::FALLBACK_SERVICE_NAME)
+    end
+  end
+
+  describe 'Configuration' do
+    let(:settings) { Datadog::Core::Configuration::Settings.new }
+
+    describe 'default values' do
+      before { setup_metrics }
+
+      it 'uses default endpoint' do
+        expect(exporter.instance_variable_get(:@uri).to_s).to eq("http://#{agent_host}:4318/v1/metrics")
+      end
+
+      it 'uses default timeout' do
+        expect(exporter.instance_variable_get(:@timeout)).to eq(10.0)
+      end
+
+      it 'uses default export interval' do
+        expect(reader.instance_variable_get(:@export_interval)).to eq(10.0)
+      end
+
+      it 'uses default export timeout' do
+        expect(reader.instance_variable_get(:@export_timeout)).to eq(7.5)
+      end
+    end
+
+    describe 'configuration priority' do
+      let(:env_vars) do
+        {
+          'OTEL_EXPORTER_OTLP_ENDPOINT' => 'http://general:4317',
+          'OTEL_EXPORTER_OTLP_PROTOCOL' => 'http/protobuf',
+          'OTEL_EXPORTER_OTLP_TIMEOUT' => '8000',
+          'OTEL_EXPORTER_OTLP_HEADERS' => 'general=value'
+        }
+      end
+
+      before { setup_metrics(env_vars) }
+
+      it 'uses the general OTLP endpoint' do
+        expect(exporter.instance_variable_get(:@uri).to_s).to eq('http://general:4317/v1/metrics')
+      end
+
+      it 'uses the general OTLP timeout' do
+        expect(exporter.instance_variable_get(:@timeout)).to eq(8.0)
+      end
+
+      it 'uses the general OTLP headers' do
+        expect(exporter.instance_variable_get(:@headers)['general']).to eq('value')
+      end
+
+      context 'when metrics-specific configs are provided' do
+        let(:env_vars) do
+          super().merge(
+            'DD_METRICS_OTEL_ENABLED' => 'true',
+            'OTEL_EXPORTER_OTLP_METRICS_ENDPOINT' => 'http://metrics:4318/v1/metrics',
+            'OTEL_EXPORTER_OTLP_METRICS_PROTOCOL' => 'http/protobuf',
+            'OTEL_EXPORTER_OTLP_METRICS_TIMEOUT' => '5000',
+            'OTEL_EXPORTER_OTLP_METRICS_HEADERS' => 'metrics=value',
+            'OTEL_METRIC_EXPORT_INTERVAL' => '4000',
+            'OTEL_METRIC_EXPORT_TIMEOUT' => '3000',
+            'OTEL_EXPORTER_OTLP_PROTOCOL' => 'grpc',
+            'OTEL_EXPORTER_OTLP_TIMEOUT' => '2000'
```

Comment:
Minor: I think this one can be removed, since it's already defined in the outer block (to 8000)

```suggestion
```